src/main.ts

```diff
@@ -99,6 +113,111 @@ async function run() {
   }
 }
 
+// Check if the source request issue is a duplicate.
+async function checkForDuplicates() {
+  const duplicateCheckEnabled = core.getInput('duplicate-check-enabled');
+
+  if (duplicateCheckEnabled !== 'true') {
+    core.info('The duplicate check is disabled');
+    return;
+  }
+
+  const payload = github.context.payload as IssuesOpenedEvent;
+
+  if (payload.action !== 'opened' || !payload.issue) {
+    core.info('Irrelevant action trigger');
+    return;
+  }
+
+  if (!payload.sender) {
+    throw new Error('Internal error, no sender provided by GitHub');
+  }
+
+  const issue = payload.issue as Issue;
+  const labelToCheck = core.getInput('duplicate-check-label', {required: true});
+  const hasTheLabel = issue.labels?.find(label => label.name === labelToCheck);
+
+  if (!hasTheLabel) {
```

Comment:
Is the intention to add the label to the source request issue template?

Reply:
Yes, in the extension repo the `duplicate-check-label` would be the `'Source request'`